### PR TITLE
ci: Use GHA instead of zeus

### PR DESCRIPTION
**Pull request by Swatinem**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,8 +29,6 @@ jobs:
           python-version: 3.7
 
       - uses: Swatinem/rust-cache@v1
-        with:
-          key: lints
```
Comment (jan-auer) [inline .github/workflows/ci.yml:33]: What changed in the action so that the key is no longer required?

**Review by jan-auer — APPROVED** [review]

**Comment by jan-auer** [comment]
Please run a test in a release branch before merging.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Swatinem) [reply .github/workflows/ci.yml:33]: It now uses the job name automatically ;-)

Comment (jan-auer) [reply .github/workflows/ci.yml:33]: Is the job name defaulted if not set explicitly, like in this case?
Don't you have to upgrade to a newer version of `rust-cache` then?

**Comment by Swatinem** [comment]
https://github.com/getsentry/symbolic/actions/runs/357922856 so there is the artifact, seems like all the python stuff in there

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Swatinem) [reply .github/workflows/ci.yml:33]: github will just use the latest commit from the `v1` branch, and it *always* attached the job name to the cache, in *addition* to the manually provided key
* before: `v0-rust-test-python-test-python-1.47.0-x86_64-unknown-linux-gnu-18bf6b4f01a6-2277f239961d34e39a31`
* after: `v0-rust-test-python-1.47.0-x86_64-unknown-linux-gnu-18bf6b4f01a6-2277f239961d34e39a31`

**Pull request merged**